app/utils/sitemap.server.ts

```diff
@@ -0,0 +1,130 @@
+        return handle.getSitemapEntries(request);
+      }
+
+      // exclude resource routes from the sitemap
+      // (these are an opt-in via the getSitemapEntries method)
+      if (!("default" in mod)) return;
+
+      const manifestEntry = manifest.routes[id];
+      if (!manifestEntry) {
+        console.warn(`Could not find a manifest entry for ${id}`);
+        return;
+      }
+      let parentId = manifestEntry.parentId;
+      let parent = parentId ? manifest.routes[parentId] : null;
+
+      let path;
+      if (manifestEntry.path) {
+        path = removeTrailingSlash(manifestEntry.path);
+      } else if (manifestEntry.index) {
+        path = "";
+      } else {
+        return;
+      }
+
+      while (parent) {
+        // the root path is '/', so it messes things up if we add another '/'
+        const parentPath = parent.path ? removeTrailingSlash(parent.path) : "";
+        path = `${parentPath}/${path}`;
+        parentId = parent.parentId;
+        parent = parentId ? manifest.routes[parentId] : null;
+      }
+
+      // we can't handle dynamic routes, so if the handle doesn't have a
+      // getSitemapEntries function, we just
+      if (path.includes(":")) {
+        return;
+      }
+
+      const entry: SitemapEntry = { route: removeTrailingSlash(path) };
+      return entry;
+    }),
+  );
+
+  const rawSitemapEntries: SitemapEntry[] = entries.flat().filter(typedBoolean);
+
+  const sitemapEntries: SitemapEntry[] = [];
+  for (const entry of rawSitemapEntries) {
+    const existingEntryForRoute = sitemapEntries.find(
+      (e) => e.route === entry.route,
+    );
+    if (existingEntryForRoute) {
+      if (!isEqual(existingEntryForRoute, entry)) {
+        console.warn(
+          `Duplicate route for ${entry.route} with different sitemap data`,
+          { entry, existingEntryForRoute },
+        );
+      }
+    } else {
+      sitemapEntries.push(entry);
+    }
+  }
+
+  return sitemapEntries;
+}
+
+function toXmlTag([key, value]: [string, string]) {
+  return value ? `<${key}>${value}</${key}>` : "";
+}
+
+function getEntry(domainUrl: string, { route, ...entry }: SitemapEntry) {
+  const tags = Object.entries(entry).map(toXmlTag);
+
+  return `<url><loc>${domainUrl}${route}</loc>${tags.join("")}</url>`;
+}
+
+export async function getSitemapXml(
+  request: Request,
+  routeModules: RouteModules<EntryRouteModule>,
+  manifest: AssetsManifest,
+) {
+  const domainUrl = getDomainUrl(request);
+
+  const sitemapEntries: SitemapEntry[] = await getSiteMapEntries(
+    request,
+    routeModules,
+    manifest,
+  );
+
+  return `
+<?xml version="1.0" encoding="UTF-8"?>
+<urlset
+  xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd"
+>
+  ${sitemapEntries.map((entry) => getEntry(domainUrl, entry)).join("")}
+</urlset>
+  `.trim();
+}
```

Comment:
:thought_balloon: _I’m just thinking out loud here. Something doesn’t necessarily have to change, but I want to make sure to share my thoughts._ [ref](https://devblogs.microsoft.com/appcenter/how-the-visual-studio-mobile-center-team-does-code-review/)
Kanskje dette nå kan puttes inn i `sitemap[.].xml`, det er jo vell bare der vi faktisk bryr oss om xml formateringen?

På en annen side så hører det kanskje sammen med den andre sitemap koden, for sitemap og xml henger jo sammen.. 🤷 
_💭 Hmm kode plassering er ikke lette greier noen ganger_ 

Hvis du ikke har noen mening bare behold slikt det er, jeg vet ikke hva som er best.

Reply:
Jeg syntes det ble bedre å flytte det 😄 [229cf02](https://github.com/capraconsulting/nettsiden/pull/6/commits/229cf02efbfc04019018198b2ec996e41ac4ebc8)